src/scry/request.cr

```diff
@@ -3,15 +3,12 @@ require "./headers"
 
 module Scry
   struct Request
-    def initialize(@io : IO)
-      @headers = Headers.new
-      @content = uninitialized String | Nil
-    end
+    getter headers = Headers.new
+    getter content : String?
 
-    def read
-      return @content if @content
+    def initialize(@io : IO)
```

Comment:
I'm wondering, do we need to store the `io` in the request? Since there's no getter for `@io` and it's not used after the initialize I think it's safe to not store it, and pass the `io` as an argument to `read_headers` & `read_content`.

Also, if you do that you can put the whole request reading in a class method `from_io(io)` that will read the headers & content then simply initialize the request with the read values.
WDYT?